<p>Sports clothing and athleisure attire is a huge industry, worth approximately <a href="https://www.statista.com/statistics/254489/total-revenue-of-the-global-sports-apparel-market/">$193 billion in 2021</a> with a strong growth forecast over the next decade! </p>
<p> In this project I dive into product data such as pricing, reviews, descriptions, and ratings, as well as revenue and website traffic, to produce recommendations for revenue growth.  </p>
<p>The database <code>sports</code>, contains five tables, with <code>product_id</code> being the primary key for all of them: </p>
<h3 id="info"><code>info</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_name</code></td>
<td><code>varchar</code></td>
<td>Name of the product</td>
</tr>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>description</code></td>
<td><code>varchar</code></td>
<td>Description of the product</td>
</tr>
</tbody>
</table>
<h3 id="finance"><code>finance</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>listing_price</code></td>
<td><code>float</code></td>
<td>Listing price for product</td>
</tr>
<tr>
<td><code>sale_price</code></td>
<td><code>float</code></td>
<td>Price of the product when on sale</td>
</tr>
<tr>
<td><code>discount</code></td>
<td><code>float</code></td>
<td>Discount, as a decimal, applied to the sale price</td>
</tr>
<tr>
<td><code>revenue</code></td>
<td><code>float</code></td>
<td>Amount of revenue generated by each product, in US dollars</td>
</tr>
</tbody>
</table>
<h3 id="reviews"><code>reviews</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_name</code></td>
<td><code>varchar</code></td>
<td>Name of the product</td>
</tr>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>rating</code></td>
<td><code>float</code></td>
<td>Product rating, scored from <code>1.0</code> to <code>5.0</code></td>
</tr>
<tr>
<td><code>reviews</code></td>
<td><code>float</code></td>
<td>Number of reviews for the product</td>
</tr>
</tbody>
</table>
<h3 id="traffic"><code>traffic</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>last_visited</code></td>
<td><code>timestamp</code></td>
<td>Date and time the product was last viewed on the website</td>
</tr>
</tbody>
</table>
<h3 id="brands"><code>brands</code></h3>
<table>
<thead>
<tr>
<th>column</th>
<th>data type</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>product_id</code></td>
<td><code>varchar</code></td>
<td>Unique ID for product</td>
</tr>
<tr>
<td><code>brand</code></td>
<td><code>varchar</code></td>
<td>Brand of the product</td>
</tr>
</tbody>
</table>


In [397]:
%%sql
postgresql:///sports

--Counting the number of non-missing entries for description, listing_price, and last_visited

SELECT 
        COUNT(*) AS total_rows,
        COUNT(i.description) AS count_description,
        COUNT(f.listing_price) AS count_listing_price,
        COUNT(t.last_visited) AS count_last_visited
        
FROM
        info AS i
JOIN 
        finance AS f
ON i.product_id = f.product_id
JOIN
        traffic AS t
ON f.product_id = t.product_id;

1 rows affected.


[(3179, 3117, 3120, 2928)]

We can see the database contains 3,179 products in total. Of the columns we previewed, only one "last_visited" is missing more than five percent of its values. Now let's turn our attention to pricing. 
How do the price points of Nike and Adidas products differ? Answering this question can help us build a picture of the company's stock range and customer market. 

In [399]:
%%sql

-- Filtering for products with a listing_price more than zero

SELECT 
        brand,
        listing_price::INTEGER,
        COUNT(f.product_id)
        
FROM finance AS f
JOIN brands AS b
ON f.product_id = b.product_id
WHERE listing_price::INTEGER > 0
GROUP BY brand, listing_price
ORDER BY listing_price::INTEGER DESC;
        

 * postgresql:///sports
77 rows affected.


[('Adidas', 300, 2),
 ('Adidas', 280, 4),
 ('Adidas', 240, 5),
 ('Adidas', 230, 8),
 ('Adidas', 220, 11),
 ('Nike', 200, 1),
 ('Adidas', 200, 8),
 ('Nike', 190, 2),
 ('Adidas', 190, 7),
 ('Nike', 180, 4),
 ('Adidas', 180, 34),
 ('Nike', 170, 14),
 ('Adidas', 170, 27),
 ('Nike', 160, 31),
 ('Adidas', 160, 28),
 ('Adidas', 150, 41),
 ('Nike', 150, 6),
 ('Adidas', 140, 36),
 ('Nike', 140, 12),
 ('Adidas', 130, 96),
 ('Nike', 130, 12),
 ('Adidas', 120, 115),
 ('Nike', 120, 16),
 ('Nike', 110, 17),
 ('Adidas', 110, 91),
 ('Nike', 100, 14),
 ('Adidas', 100, 72),
 ('Adidas', 96, 2),
 ('Nike', 95, 1),
 ('Nike', 90, 13),
 ('Adidas', 90, 89),
 ('Adidas', 86, 7),
 ('Nike', 85, 5),
 ('Adidas', 85, 1),
 ('Adidas', 80, 322),
 ('Nike', 80, 16),
 ('Nike', 79, 1),
 ('Adidas', 76, 149),
 ('Adidas', 75, 1),
 ('Nike', 75, 7),
 ('Adidas', 70, 87),
 ('Nike', 70, 4),
 ('Adidas', 66, 102),
 ('Nike', 65, 1),
 ('Adidas', 63, 1),
 ('Nike', 60, 2),
 ('Adidas', 60, 211),
 ('Adidas', 56, 174),
 ('Adidas', 55, 2),
 ('Adidas', 53, 43),
 ('Nike', 50, 5),
 ('Adidas', 50, 183),
 ('Adidas', 48, 42),
 ('Nike', 48, 1),
 ('Adidas', 46, 163),
 ('Nike', 45, 3),
 ('Adidas', 45, 1),
 ('Adidas', 43, 51),
 ('Adidas', 40, 81),
 ('Nike', 40, 1),
 ('Adidas', 38, 24),
 ('Adidas', 36, 25),
 ('Adidas', 33, 24),
 ('Adidas', 30, 37),
 ('Nike', 30, 2),
 ('Adidas', 28, 38),
 ('Adidas', 27, 18),
 ('Adidas', 25, 28),
 ('Adidas', 23, 1),
 ('Adidas', 20, 8),
 ('Adidas', 18, 4),
 ('Adidas', 16, 4),
 ('Adidas', 15, 27),
 ('Adidas', 13, 27),
 ('Adidas', 12, 1),
 ('Adidas', 10, 11),
 ('Adidas', 9, 1)]

It turns out there are 77 unique prices for the products in our database, which makes the output of the last query quite difficult to analyze. 
I'll build on the previous query by assigning labels to different price ranges, grouping by 'brand' and 'label'. We will also include the total 'revenue' for each price range and 'brand'.

In [401]:
%%sql

-- Creating labels for products based on their price range, aliasing as price_category

SELECT 
        brand,
        COUNT(f.product_id),
        SUM(revenue) AS total_revenue,
        CASE WHEN listing_price::INTEGER < 42 THEN 'Budget'
            WHEN listing_price::INTEGER >= 42 AND listing_price::INTEGER < 74 THEN 'Average'
            WHEN listing_price::INTEGER >= 74 AND listing_price::INTEGER < 129 THEN 'Expensive'
            WHEN listing_price::INTEGER >= 129 THEN 'Elite'
            END AS price_category
        
FROM finance AS f
JOIN brands AS b
ON f.product_id = b.product_id
WHERE brand IS NOT NULL
GROUP BY brand, price_category
ORDER BY total_revenue DESC;


 * postgresql:///sports
8 rows affected.


[('Adidas', 849, 4626980.069999999, 'Expensive'),
 ('Adidas', 1060, 3233661.060000001, 'Average'),
 ('Adidas', 307, 3014316.8299999987, 'Elite'),
 ('Adidas', 359, 651661.1200000002, 'Budget'),
 ('Nike', 357, 595341.0199999992, 'Budget'),
 ('Nike', 82, 128475.59000000003, 'Elite'),
 ('Nike', 90, 71843.15000000004, 'Expensive'),
 ('Nike', 16, 6623.5, 'Average')]

Interestingly, grouping products by brand and price range allows us to see that Adidas items generate more total revenue regardless of price category! Specifically, "Elite" Adidas products priced $129 or more typically generate the highest revenue, so the company can potentially increase revenue by shifting their stock to have a larger proportion of these products.
Note that we have been looking at 'listing_price' so far. The 'listing_price' may not be the price that the product is ultimately sold for. To understand 'revenue' better, let's take a look at the 'discount', which is the percent reduction in the 'listing_price' when the product is actually sold. I would like to know whether there is a difference in the amount of 'discount' offered between brands, as this could be influencing 'revenue'.

In [403]:
%%sql

-- Selecting brand and average_discount as a percentage

SELECT 
        brand,
        AVG(discount)*100 AS average_discount
FROM brands AS b
JOIN finance AS f
ON b.product_id = f.product_id
WHERE brand IS NOT NULL
GROUP BY brand;

 * postgresql:///sports
2 rows affected.


[('Nike', 0.0), ('Adidas', 33.452427184465606)]

Strangely, no 'discount' is offered on Nike products! In comparison, not only do Adidas products generate the most revenue, but these products are also heavily discounted!
To improve revenue further, the company could try to reduce the amount of discount offered on Adidas products, and monitor sales volume to see if it remains stable. Alternatively, it could try offering a small discount on Nike products. This would reduce average revenue for these products, but may increase revenue overall if there is an increase in the volume of Nike products sold.
Now I will explore whether relationships exist between the columns in the database. I will check the strength and direction of a correlation between 'revenue' and 'reviews'.

In [405]:
%%sql

-- Calculating the correlation between reviews and revenue as review_revenue_corr

SELECT
        CORR(reviews, revenue) AS review_revenue_corr
FROM reviews AS r
JOIN finance AS f
ON r.product_id = f.product_id;

 * postgresql:///sports
1 rows affected.


[(0.6518512283481301,)]

Interestingly, there is a positive correlation between 'revenue' and 'reviews'. This means, potentially, if we can get more reviews on the company's website, it may increase sales of those items with a larger number of reviews. 
Perhaps the length of a product's 'description' might influence a product's 'rating' and 'reviews'. If so, the company can produce content guidelines for listing products on their website and test if this influences 'revenue'. 

In [407]:
%%sql

-- Calculating description_length, average_rating
-- Filtering for products without missing values for description


SELECT TRUNC(LENGTH(i.description), -2) AS description_length,
    ROUND(AVG(r.rating::numeric), 2) AS average_rating
FROM info AS i
INNER JOIN reviews AS r 
    ON i.product_id = r.product_id
WHERE i.description IS NOT NULL
GROUP BY description_length
ORDER BY description_length;

 * postgresql:///sports
7 rows affected.


[(Decimal('0'), Decimal('1.87')),
 (Decimal('100'), Decimal('3.21')),
 (Decimal('200'), Decimal('3.27')),
 (Decimal('300'), Decimal('3.29')),
 (Decimal('400'), Decimal('3.32')),
 (Decimal('500'), Decimal('3.12')),
 (Decimal('600'), Decimal('3.65'))]

Unfortunately, there doesn't appear to be a clear pattern between the length of a product's 'description' and its 'rating'.
As we know a correlation exists between 'reviews' and 'revenue', one approach the company could take is to run experiments with different sales processes encouraging more reviews from customers about their purchases, such as by offering a small discount on future purchases.
I'll take a look at the volume of 'reviews' by month to see if there are any trends or gaps to exploit.

In [409]:
%%sql

SELECT 
        brand,
        DATE_PART('month', last_visited) AS month,
        COUNT(r.product_id) AS num_reviews
FROM traffic AS t
JOIN reviews AS r
ON t.product_id = r.product_id
JOIN brands AS b
ON r.product_id = b.product_id
WHERE brand IS NOT NULL
        AND last_visited IS NOT NULL
GROUP BY brand, month
ORDER BY brand, month;

 * postgresql:///sports
24 rows affected.


[('Adidas', 1.0, 253),
 ('Adidas', 2.0, 272),
 ('Adidas', 3.0, 269),
 ('Adidas', 4.0, 180),
 ('Adidas', 5.0, 172),
 ('Adidas', 6.0, 159),
 ('Adidas', 7.0, 170),
 ('Adidas', 8.0, 189),
 ('Adidas', 9.0, 181),
 ('Adidas', 10.0, 192),
 ('Adidas', 11.0, 150),
 ('Adidas', 12.0, 190),
 ('Nike', 1.0, 52),
 ('Nike', 2.0, 52),
 ('Nike', 3.0, 55),
 ('Nike', 4.0, 42),
 ('Nike', 5.0, 41),
 ('Nike', 6.0, 43),
 ('Nike', 7.0, 37),
 ('Nike', 8.0, 29),
 ('Nike', 9.0, 28),
 ('Nike', 10.0, 47),
 ('Nike', 11.0, 38),
 ('Nike', 12.0, 35)]

Looks like product reviews are highest in the first quarter of the calendar year, so there is scope to run experiments aiming to increase the volume of reviews in the other nine months.
So far, I have been primarily analyzing Adidas vs Nike products. Now, let's switch the attention to the type of products being sold. Let's find out how much of the company's stock consists of footwear products and the median 'revenue' generated by these items.

In [411]:
%%sql

-- Creating the footwear CTE, containing description and revenue
-- Filtering footwear for products with a description containing %shoe%, %trainer, or %foot%


WITH footwear AS (
    SELECT
            description,
            revenue
    FROM info AS i
    JOIN finance AS f
    ON i.product_id = f.product_id
    WHERE description iLIKE '%shoe%'
    OR description iLIKE '%trainer%'
    OR description iLIKE '%foot%'
    AND description IS NOT NULL
)

SELECT COUNT(*) AS num_footwear_products, 
    percentile_disc(0.5) WITHIN GROUP (ORDER BY revenue) AS median_footwear_revenue
FROM footwear;


 * postgresql:///sports
1 rows affected.


[(2700, 3118.36)]

Recall from the first query that there are 3,117 products without missing values for 'description'. Of those, 2,700 are footwear products, which accounts for around 85% of the company's stock. They also generate a median revenue of over $3000 dollars.
This is interesting, but there is no point of reference for whether footwear's 'median_revenue' is good or bad compared to other products. So let's examine how this differs to clothing products. I will re-use 'footwear', add a filter afterwards to count the number of products and 'median_revenue' of products that are not in 'footwear'.

In [413]:
%%sql


WITH footwear AS
(
    SELECT i.description, f.revenue
    FROM info AS i
    INNER JOIN finance AS f 
        ON i.product_id = f.product_id
    WHERE i.description ILIKE '%shoe%'
        OR i.description ILIKE '%trainer%'
        OR i.description ILIKE '%foot%'
        AND i.description IS NOT NULL
)

SELECT COUNT(i.*) AS num_clothing_products, 
    percentile_disc(0.5) WITHIN GROUP (ORDER BY f.revenue) AS median_clothing_revenue
FROM info AS i
INNER JOIN finance AS f on i.product_id = f.product_id
WHERE i.description NOT IN (SELECT description FROM footwear);

 * postgresql:///sports
1 rows affected.


[(417, 503.82)]

Looks like 'footwear' products median revenue is 6 times bigger than the one of 'clothing' products. So there's a point to analyze 'footwear' products closely and base the optimization tips on this category. Small change for better in 'footwear' will have a bigger value to the overall results than the same type of change for clothing products.